tests/cmaes_test.cpp

```diff
@@ -53,6 +65,10 @@ TEST_CASE("CMAESLogisticRegressionFMatTest", "[CMAESTest]")
  */
 TEST_CASE("ApproxCMAESLogisticRegressionFMatTest", "[CMAESTest]")
 {
-  ApproxCMAES<> cmaes(0, -1, 1, 32, 200, 1e-3);
+  arma::fmat lb(1, 1, arma::fill::value(-1));
+  arma::fmat ub(1, 1, arma::fill::value(1));
+  BoundaryBoxConstraint b(lb, ub);
+
+  ApproxCMAES<BoundaryBoxConstraint<arma::fmat>> cmaes(0, b, 32, 200, 1e-3);
```

Comment:
When you add the deprecated constructor, we should at least add one test that uses the deprecated constructor (and we can point out in a comment that the test itself can be removed when ensmallen 3.x is released, as the constructor will go away).

Reply:
Yes, makes sense.